src/game/AI/ScriptDevAI/scripts/kalimdor/dustwallow_marsh.cpp

```diff
@@ -1498,6 +1511,462 @@ UnitAI* GetAI_npc_smolderwing(Creature* pCreature)
+
+    SAY_BACK_TO_CORNERS = 1636, //Criton
+
+    SAY_FOR_THE_RED = 1778,
+    SAY_FOR_THE_BLUE = 1777,
+
+    SAY_FIRST_COMBATANTS = 1771,
+    SAY_FIGHT_INTRO = 1643,
+    SAY_FIGHT_START = 1641,
+
+    SAY_EDWARD_LOSS = 1760,
+    SAY_KAHIL_LOSS = 1762,
+    SAY_NARRISHA_LOSS = 1763,
+
+    SAY_JARAD_LOSS = 1766,
+    SAY_LANA_LOSS = 1767,
+    SAY_TARK_LOSS = 1768,
+
+    SAY_RED_MOTIVATION_WIN = 1769,
+    SAY_RED_MOTIVATION_LOSS = 1770,
+
+    SAY_BLUE_MOTIVATION_LOSS = 1765,
+    SAY_BLUE_MOTIVATION_WIN = 1764,
+
+    SAY_RED_FIRST = 1776,
+    SAY_BLUE_FIRST = 1775,
+};
+
+enum CombatantActions
+{
+    COMBATANT_CHECK_HP,
+    COMBATANT_BLUE_WIN,
+    COMBATANT_RED_WIN,
+    COMBATANT_START_POSITION,
+    COMBATANT_START_SPAR,
+    COMBATANT_GET_ALL,
+    COMBATANT_SALUTE,
+    COMBATANT_SWITCH,
+    COMBATANT_SEPARATE,
+    COMBATANT_INTRO,
+    COMBATANT_MOTIVATION,
+    COMBATANT_ACTION_MAX,
+};
+
+static std::unordered_map<uint32, const Position> startingPositions ({
+    {BLUE_TEAM, {-3643.574707, -4505.717773, 9.462863, 3.855592}},
+    {RED_TEAM, {-3652.405273, -4514.126953, 9.462863, 0.717925}}
+});
+
+static const uint32 theramore_blue_team[] = {NPC_GUARD_EDWARD, NPC_GUARD_KAHIL, NPC_GUARD_NARRISHA};
+static const uint32 theramore_red_team[] = {NPC_GUARD_JARAD, NPC_GUARD_LANA, NPC_GUARD_TARK};
+
+struct npc_theramore_spar_controller : public CombatAI
+{
+    npc_theramore_spar_controller(Creature* creature) : CombatAI(creature, COMBATANT_ACTION_MAX)
+    {
+        AddCustomAction(COMBATANT_CHECK_HP, true, [&](){
+            CheckCombatantHP();
+        });
+        AddCustomAction(COMBATANT_BLUE_WIN, true, [&](){
+            HandleBlueWin();
+        });
+        AddCustomAction(COMBATANT_RED_WIN, true, [&](){
+            HandleRedWin();
+        });
+        AddCustomAction(COMBATANT_START_SPAR, true, [&](){
+            StartSparring();
+        });
+        AddCustomAction(COMBATANT_START_POSITION, true, [&](){
+            MoveIntoPosition();
+        });
+        AddCustomAction(COMBATANT_SALUTE, true, [&](){
+            CombatantSalute();
+        });
+        AddCustomAction(COMBATANT_INTRO, true, [&](){
+            CombatantIntro();
+        });
+        AddCustomAction(COMBATANT_SWITCH, true, [&](){
+            SwitchMembers();
+        });
+        AddCustomAction(COMBATANT_SEPARATE, true, [&](){
+            CombatantSeparate();
+        });
+        AddCustomAction(COMBATANT_GET_ALL, true, [&](){
+            GetCombatants();
+        });
+        AddCustomAction(COMBATANT_MOTIVATION, true, [&](){
+            HandleMotivation();
+        });
+        Reset();
+    }
+
+    std::unordered_map<uint32, GuidList> teams;
+    std::unordered_map<uint32, ObjectGuid> medics;
+    std::unordered_map<uint32, ObjectGuid> active;
+
+    uint32 m_uiWinningTeam;
+
+    void Reset() override
+    {
+        ResetTimer(COMBATANT_GET_ALL, 2000u);
+        ResetTimer(COMBATANT_START_POSITION, 10000u);
```

Comment:
Need to clear maps before re-starting the event

Reply:
Firstly, you are not calling CombatAI::Reset()
Secondly, if those timers are meant to be in start, pass them directly to AddCustomAction, which has support for it and you no longer have to override Reset at all